392 d4hj implement in panel land ownership data notification

ms0ur1s opened the pull request:
#### What? Why?

Fixes #392

The need to inform users that the 'Land Ownership' data may be out of date. 

Introduces a warning notification component to the 'Land Ownership' section in the left pane, alerting users that the data may be outdated. The implementation includes a new reusable component, session-based dismissal logic, and associated styling for notifications.

#### Code-specific details (for Reviewer)

**Feature Addition:**
- Added a `LeftPaneNotification` React component for showing and dismissing notifications, with session storage used to remember dismissed state.
- Integrated the dismissible warning notification into the 'Land Ownership' section of the left pane, displaying a message and external link for data verification.

**Styling and Assets:**
- Created new SCSS styles for the notification component, including variants for warnings and left-pane placement, and embedded SVG icons for warning and close actions.
- Added new color variables for notification backgrounds and accents in `_variables.scss`.
- Imported the new notification styles into the main stylesheet.

**Integration:**
- Imported the new `LeftPaneNotification` component into the left pane code.

***

#### What should we test? (for QA)

- Open the Data Layers pane
- Check that the notification appears at the top of the Land Ownership section
- Check the 'Land Registry search' link correctly opens in a new tab
- Click the 'x' in the top right of the notification to close it
- Open dev tools
- Click on the 'Application' tab
- Ensure that the Key `lx.landOwnershipNotification.hidden` and the Value `true` appear in 'Session storage'
- Return to the browser
- Refresh the relevant tab and ensure that the notification doesn't appear
- Open another tab
- Navigate to localhost:8080/app
- Repeat the first and second steps

ms0ur1s committed:
Implement land ownership data notification component and desktop styles

ms0ur1s committed:
Refactor land ownership notification component to improve state management and readability

ms0ur1s committed:
Refactor land ownership notification component to use updated styles and class names

ms0ur1s committed:
Replace LandOwnershipNotification with DismissibleNotification

Now a re-usable component

ms0ur1s committed:
Renamed DismissibleNotification to LeftPaneNotification

rogup commented on src/assets/styles/_notifications.scss:
```diff
@@ -0,0 +1,61 @@
+        display: block;
+        width: 8px;
+        height: 8px;
+        background-image: url('data:image/svg+xml,<svg viewBox="0 0 8 8" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1.13862.192782c-.260317-.2543302-.683069-.2543302-.943384 0-.2603147.254331-.2603147.667364 0 .921698L3.05662 3.90805.197319 6.70365c-.2603154.25433-.2603154.66736 0 .92169.260315.25433.683066.25433.943381 0L4 4.82974l2.86138 2.79357c.26032.25433.68307.25433.94338 0 .2 ...
```
@ms0ur1s This is more of a question than a comment since I'm curious - how do you generate the numbers in these paths?

ms0ur1s replied:
I just copied the svg code of the icon in figma, optimised it [here](https://svgomg.net/) and generated the background css [here](https://www.svgbackgrounds.com/tools/svg-to-css/).

rogup commented on src/components/left-pane/left-pane-notification/LeftPaneNotification.js:
```diff
@@ -0,0 +1,49 @@
+
+const DismissibleNotification = ({
+  storageKey,
+  notificationType,
```
Maybe include a JSDoc above this component to say what values notificationType can take

rogup commented on src/components/left-pane/left-pane-notification/LeftPaneNotification.js:
```diff
@@ -0,0 +1,49 @@
+import React, { useState } from "react";
+
+const DismissibleNotification = ({
```
A minor thing, but I think we should call this component LeftPaneNotification since we generally follow this convention of keeping the component and file names the same. It does sometimes help with refactoring e.g when renaming things in bulk
Also, this component does have Left pane-specific styling hardcoded into it

ms0ur1s replied:
Great spot, I renamed the component file name but forgot to rename the component itself.

rogup commented on src/components/left-pane/LeftPaneLandData.js:
```diff
@@ -118,6 +119,23 @@ const LeftPaneLandData = ({ open, active, onClose }) => {
       </DataLayersContainer>
       <DataLayersContainer title={"Land Ownership"}>
+        <DismissibleNotification
+          storageKey="lx.landOwnershipNotification.hidden"
```
Rather than storage key, this component could just expose 'notificationId' as an input e.g. 'landOwnershipData'. Then inside the component it could form a storageKey such as `lx.notification.${notificationId}.hidden`
This means that when we use the component elsewhere in the app, we don't need to know about its internal workings and the fact is uses session storage.

ms0ur1s replied:
Cool, makes total sense.

rogup reviewed (commented):
Nice work! It looks fully functional, I've just left a few very minor comments as suggestions for best practice

ms0ur1s committed:
Review resolution #1
Rename DismissibleNotification to LeftPaneNotification, storageKey to notificationId, add JSDoc

ms0ur1s commented:
Sorry @rogup I resolved three of the comments before you could review them.

ms0ur1s committed:
Added rel attribute to Land Registry search link

Removed by accident

rogup commented:
@ms0ur1s That's all good! It's fine for you to resolve comments that you've addressed. Sending them back for double-checking isn't necessary and makes the whole process longer, unless you have further clarification questions :)
